cmake/KokkosCore_config.h.in

```diff
@@ -55,45 +54,38 @@
 
 #cmakedefine KOKKOS_IMPL_CUDA_CLANG_WORKAROUND
 
+#cmakedefine KOKKOS_COMPILER_CUDA_VERSION @KOKKOS_COMPILER_CUDA_VERSION@
 
 #cmakedefine KOKKOS_ARCH_NONE
-#cmakedefine KOKKOS_ARCH_AMDAVX
+#cmakedefine KOKKOS_ARCH_SSE42
 #cmakedefine KOKKOS_ARCH_ARMV80
 #cmakedefine KOKKOS_ARCH_ARMV81
-#cmakedefine KOKKOS_ARCH_ARMV8_THUNDERX
-#cmakedefine KOKKOS_ARCH_ARMV8_TX2
-#cmakedefine KOKKOS_ARCH_SSE42
-#cmakedefine KOKKOS_ARCH_WSM
-#cmakedefine KOKKOS_ARCH_SNB
+#cmakedefine KOKKOS_ARCH_AMD_AVX2
 #cmakedefine KOKKOS_ARCH_AVX
-#cmakedefine KOKKOS_ARCH_HSW
-#cmakedefine KOKKOS_ARCH_BDW
 #cmakedefine KOKKOS_ARCH_AVX2
-#cmakedefine KOKKOS_ARCH_SKX
 #cmakedefine KOKKOS_ARCH_AVX512XEON
 #cmakedefine KOKKOS_ARCH_KNC
 #cmakedefine KOKKOS_ARCH_AVX512MIC
-#cmakedefine KOKKOS_ARCH_KNL
-#cmakedefine KOKKOS_ARCH_BGQ
```

Comment:
Please comment here (on gh) on these line changes.

Reply:
`Makefile.kokkos` does not have a KNL define. It defines `KOKKOS_ARCH_AVX512MIC` instead. BGQ doesn't define anything in `Makefile.kokkos`.